module/Activity/src/Activity/Service/ActivityQuery.php

```diff
@@ -222,6 +221,7 @@ public function getUpcomingActivities($category = null, $user = null)
 
         $activityMapper = $this->getServiceManager()->get('activity_mapper_activity');
         if ($category === 'my') {
+            $user = $this->getServiceLocator()->get('user_service_user')->getIdentity();
             return $activityMapper->getUpcomingActivitiesForMember($user);
```

Comment:
It would make more sense to directly call `getUpcomingActivitiesForMember` from the controller, especially since getting the user identity is usually done at the controller level.

Reply:
I would instead argue that it should be part of the Service level.